validator/client/src/main/java/tech/pegasys/teku/validator/client/loader/ValidatorLoader.java

```diff
@@ -54,20 +54,23 @@ public class ValidatorLoader {
   private final GraffitiProvider graffitiProvider;
   private final Optional<DataDirLayout> maybeDataDirLayout;
   private final SlashingProtectionLogger slashingProtectionLogger;
+  private final boolean allowNoLoadedKeys;
 
   private ValidatorLoader(
       final List<ValidatorSource> validatorSources,
       final Optional<ValidatorSource> mutableLocalValidatorSource,
       final Optional<ValidatorSource> mutableExternalValidatorSource,
       final GraffitiProvider graffitiProvider,
       final Optional<DataDirLayout> maybeDataDirLayout,
-      final SlashingProtectionLogger slashingProtectionLogger) {
+      final SlashingProtectionLogger slashingProtectionLogger,
+      final boolean allowNoLoadedKeys) {
```

Comment:
This would likely be a lot simpler at the ValidatorClientService layer rather than here...

basically I'd suggest putting a check in `ValidatorClientService.create` immediately prior to `return new ValidatorClientService` something like:
```
if (validatorLoader.getOwnedValidators().getActiveValidators().size() == 0) {
}
```

I'm not sure we're looking at only validators with a local signer, that's probably over-complicating things and reducing what it can be used for...